test/handlers/ProveGatewayHandler/persist.test.ts

```diff
@@ -0,0 +1,125 @@
+      gatewayAddress,
+      chain,
+      gatewayType,
+      remoteGatewayAddress,
+      tokenAddress,
+      anchorAddress,
+      bounty,
+      activation,
+      lastRemoteGatewayProvenBlockHeight,
+      createdAt,
+      updatedAt,
+    );
+    await config.repos.gatewayRepository.save(
+      gateway,
+    );
+  });
+
+  it('should persist successfully', async (): Promise<void> => {
+    const updatedLastRemoteProvenBlockHeight = new BigNumber('10');
+    const proveGatewayTransactions = [{
+      id: '1',
+      _gateway: gatewayAddress,
+      _blockHeight: updatedLastRemoteProvenBlockHeight,
+      _storageRoot: Utils.sha3('1'),
+      _wasAlreadyProved: false,
+      contractAddress: '0x00000000000000000000000000000000000000011',
+      blockNumber: new BigNumber('100'),
+      uts: new BigNumber('1111'),
+    }];
+
+    const handler = new ProveGatewayHandler(config.repos.gatewayRepository);
+    await handler.persist(proveGatewayTransactions);
+
+    const updatedGateway = await config.repos.gatewayRepository.get(gatewayAddress);
+    assert.deepEqual(
+      updatedGateway && updatedGateway.lastRemoteGatewayProvenBlockHeight,
+      updatedLastRemoteProvenBlockHeight,
+      'Error updating GatewayProven block height in Gateway model.',
+    );
+  });
+
+  it('It should throw error when Gateway doesn"t exist', async (): Promise<void> => {
+    const updatedLastRemoteProvenBlockHeight = new BigNumber('10');
+    const invalidGateway = 'invalidGateway';
+    const proveGatewayTransactions = [{
+      id: '1',
+      _gateway: 'invalidGateway',
+      _blockHeight: updatedLastRemoteProvenBlockHeight,
+      _storageRoot: Utils.sha3('1'),
+      _wasAlreadyProved: false,
+      contractAddress: '0x00000000000000000000000000000000000000011',
+      blockNumber: new BigNumber('100'),
+      uts: new BigNumber('1111'),
+    }];
+
+    const handler = new ProveGatewayHandler(config.repos.gatewayRepository);
+
+    assert.isRejected(
+      handler.persist(proveGatewayTransactions),
+      `Cannot find record for gateway: ${invalidGateway}`,
+      'Invalid Gateway record.',
+    );
+  });
+
+  it('should not update when received provenGatewayBlock height is less than already updated'
+    + ' provenGatewayBlock height.', async (): Promise<void> => {
+    const updatedLowerLastRemoteProvenBlockHeight = new BigNumber('1');
+    const proveGatewayTransactions = [{
+      id: '1',
+      _gateway: gatewayAddress,
+      _blockHeight: updatedLowerLastRemoteProvenBlockHeight,
+      _storageRoot: Utils.sha3('1'),
+      _wasAlreadyProved: false,
+      contractAddress: '0x00000000000000000000000000000000000000011',
+      blockNumber: new BigNumber('100'),
+      uts: new BigNumber('1111'),
+    }];
+
+    const handler = new ProveGatewayHandler(config.repos.gatewayRepository);
+    await handler.persist(proveGatewayTransactions);
+
+    const updatedGateway = await config.repos.gatewayRepository.get(gatewayAddress);
+    assert.deepEqual(
```

Comment:
It's being checked that updated lastRemoteGatewayProvenBlockHeight and original lastRemoteGatewayProvenBlockHeight should be same. Even if it would have call save,  it's strictly checking block height should be same.